CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/panes/DataAccessSearchProviderNGTest.java

```diff
@@ -0,0 +1,324 @@
+        doNothing().when(qpPane).expandPlugin(Mockito.eq("SelectTopN"));
+
+        // Mock the static method getQueryPhasePane to return the mocked QueryPhasePane
+        try (MockedStatic<DataAccessPane> mockedStatic = Mockito.mockStatic(DataAccessPane.class)) {
+            mockedStatic.when(() -> DataAccessPane.getQueryPhasePane(Mockito.anyObject())).thenReturn(qpPane);
+
+            // Return a new tab when the DataAccessPane mock is prompted for getCurrentTab
+            when(daPane.getCurrentTab()).thenReturn(new Tab());
+
+            // Mock the static method getDataAccessPane and return the mocked DataAccessPane
+            try (MockedStatic<DataAccessUtilities> mockedStatic2 = Mockito.mockStatic(DataAccessUtilities.class)) {
+                mockedStatic2.when(() -> DataAccessUtilities.getDataAccessPane()).thenReturn(daPane);
+
+                try {
+                    PluginDisplayer pd = new PluginDisplayer("SelectTopN");
+                    pd.run();
+                } catch (final IllegalStateException ex) {
+                    // Catch the exception which happens when executing Platform thread in standalone tests
+                    // Fail for any other exception
+                }
+
+                // Verify that getCurrentTab was succcessfully called.
+                verify(daPane, times(1)).getCurrentTab();
+
+                // Verify that getQueryPhasePane was succcessfully called.
+                mockedStatic.verify(() -> DataAccessPane.getQueryPhasePane(Mockito.anyObject()));
+            }
+        }
+    }
+
+    /**
+     * Test of run method, of class PluginDisplayer. Testing when tab is null
+     */
+    @Test
+    public void testRun2() {
+        System.out.println("testRun2 testing null tab returned");
+
+        // Setting up mocks
+        MockitoAnnotations.initMocks(this);
+        daPane = mock(DataAccessPane.class);
+        notifDisplayer = mock(NotificationDisplayer.class);
+        final Notification notif = new Notification() {
+            @Override
+            public void clear() {
+                // do nothing, used for testing.
+            }
+        };
+
+        // Setting up a mock for DataAccessUtilities to return null when fetching the pane.
+        try (MockedStatic<DataAccessUtilities> mockedStatic1 = Mockito.mockStatic(DataAccessUtilities.class)) {
+            when(daPane.getCurrentTab()).thenReturn(null);
+            mockedStatic1.when(() -> DataAccessUtilities.getDataAccessPane()).thenReturn(daPane);
+
+            // Mock the static method getDefault() to return the mock of NotificationDisplayer
+            try (MockedStatic<NotificationDisplayer> mockedStatic3 = Mockito.mockStatic(NotificationDisplayer.class)) {
+                when(notifDisplayer.notify("Data Access view", UserInterfaceIconProvider.WARNING.buildIcon(16, ConstellationColor.DARK_ORANGE.getJavaColor()), PluginDisplayer.STEP_STRING, null)).thenReturn(notif);
```

Comment:
To make things easier to read can you extract out `UserInterfaceIconProvider.WARNING.buildIcon(16, ConstellationColor.DARK_ORANGE.getJavaColor()` as its used twice here?

Ditto `testRun3`

Reply:
Easy!